### PR TITLE
[mergebot] Update merge bot messages to explain flags

### PR DESCRIPTION
**Pull request by zengk95**
## FORCE
@pytorchbot successfully started a merge job. Check the current status [here](None).
The merge job was triggered with the force (-f) flag. This means your change will be merged **immediately**, bypassing any CI checks (ETA: 1-5 minutes). If this is not the intended behavior, feel free to use some of the other merge options in the [wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands).
Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours) with feedback or questions!
## GREEN
@pytorchbot successfully started a merge job. Check the current status [here](None).
The merge job was triggered with the green (-g) flag. This means that your change will be merged once all checks on your PR have passed (ETA: 0-4 Hours). If this is not the intended behavior, feel free to use some of the other merge options in the [wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands).
Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours) with feedback or questions!
## LAND CHECKS
@pytorchbot successfully started a merge job. Check the current status [here](None).
The merge job was triggered with the land checks (-l) flag. If you did not specify this flag yourself,  you are likely enrolled in the [land checks rollout](https://github.com/pytorch/test-infra/blob/main/torchci/lib/bot/rolloutUtils.ts#L1-L34). This means that your change will be merged once all checks on your PR and the land checks have passed (**ETA 4 Hours**). If you need to coordinate lands between different changes and cannot risk a land race, please add the `ciflow/trunk` label to your PR and wait for signal to complete, and then land your changes in proper order. Having `trunk`, `pull`, and `Lint` pre-run on a PR will bypass land checks and the ETA should be immediate. If this is not the intended behavior, feel free to use some of the other merge options in the [wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands).
Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours) with feedback or questions!
## LAND CHECKS
@pytorchbot successfully started a merge job. Check the current status [here](None).
The merge job was triggered with the land checks (-l) flag. If you did not specify this flag yourself,  you are likely enrolled in the [land checks rollout](https://github.com/pytorch/test-infra/blob/main/torchci/lib/bot/rolloutUtils.ts#L1-L34). This means that your change will be merged once all checks on your PR have passed since you have added the `ciflow/trunk` label to your PR (ETA 0-4 Hours). If this is not the intended behavior, feel free to use some of the other merge options in the [wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands).
Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours) with feedback or questions!
## NORMAL
@pytorchbot successfully started a merge job. Check the current status [here](None).
The merge job was triggered without a flag. This means that your change will be merged once all checks on your PR have passed (ETA: 0-4 Hours). If this is not the intended behavior, feel free to use some of the other merge options in the [wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands).
Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours) with feedback or questions!
## Revert Message
@pytorchbot successfully started a revert job. Check the current status [here](None).
Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours) with feedback or questions!
## TROUBLESHOOTING 
 If you believe this is an error, you can use the old behavior with `@pytorchbot merge -g` (optionally with the `ciflow/trunk` to get land checks) or use `@pytorchbot merge -f "some reason here"`. For more information, see the [bot wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands). 
Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours) with feedback or questions!

### CODE REVIEW

File: .github/scripts/trymerge.py
Comment (janeyx99) [inline .github/scripts/trymerge.py:1211]: I think we may want this before creating the land time check branch so that we don't waste resources. We can just do if blah and not force

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [inline .github/scripts/trymerge_explainer.py:69]:
```suggestion
    def _get_flag_explanation_message(self) -> str:
```

nit

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [inline .github/scripts/trymerge_explainer.py:64]:
```suggestion
                f" and land check "
                + f"progress [here](https://hud.pytorch.org/{self.org}/{self.project}/commit/{commit})."
                + f"\n For other forms of merge, please refer to our [wiki]({BOT_COMMANDS_WIKI})."
```

Comment (janeyx99) [reply .github/scripts/trymerge_explainer.py:64]: The land check link is confusing

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [inline .github/scripts/trymerge_explainer.py:72]:
```suggestion
                "This means your PR will be merged immediately, bypassing any CI signals."
```
Let's be consistent with wording

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [inline .github/scripts/trymerge_explainer.py:75]:
```suggestion
            return "This means that your PR will be merged once all CI signals triggered on your PR have passed, including any triggered through the use of adding ciflow labels."
```

Comment (janeyx99) [reply .github/scripts/trymerge_explainer.py:75]: Let's be as clear as possible.

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [inline .github/scripts/trymerge_explainer.py:79]:
```suggestion
                    f"have run since you have added the {CIFLOW_TRUNK_LABEL} to your PR."
```

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [inline .github/scripts/trymerge_explainer.py:83]:
```suggestion
                    "and the land checks have run (ETA 4 Hours)."
```
Do we have access to the land checks link here? Can we tell people to track their progress there too?

File: .github/scripts/trymerge.py
Comment (zengk95) [reply .github/scripts/trymerge.py:1211]: I guess but it adds a bit more code since we need to call print_merge_message twice. land checks and force are mutually exclusive rn (at least in test-infra) so it'll never create a land check branch if force is true.

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [reply .github/scripts/trymerge_explainer.py:72]: we can also call them CI checks

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [reply .github/scripts/trymerge_explainer.py:75]: signals/checks let's be consistent here too

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [reply .github/scripts/trymerge_explainer.py:83]: Maybe that will get rolled up into the comment already, not sure what the whole comment looks like for each--is there a place I can see the combined versions?

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [inline .github/scripts/trymerge_explainer.py:96]:
```suggestion
                return "Since your PR has a ciflow label, we will wait for all checks to be green."
```
missing green?

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [inline .github/scripts/trymerge_explainer.py:104]:
```suggestion
        explanation_message = self._get_flag_explaination_message()
```
nit

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [reply .github/scripts/trymerge_explainer.py:66]: Okay now that I've read more code, the "for other forms of merge..." should go at the end of the whole msg. We should also link people to the DevX Q&A group or tell people to tag @pytorch-dev-infra when something is wrong so people can report bugs externally and internally.

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [inline .github/scripts/trymerge_explainer.py:121]: We should have a general troubleshooting message where they can report bugs. See comment earlier about sending people to the DevX group/tagging us?

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [inline .github/scripts/trymerge_explainer.py:58]: somewhere we should be clear like "or you have been enrolled into having -l as default." Unsure where that shoudl go

Comment (zengk95) [reply .github/scripts/trymerge_explainer.py:61]: We don't know if the user was enrolled into -l by default here :|. We only know in the test-infra and I'm planning on just letting the user know in a separate PR that they were enrolled into it.

**Review by huydhn — APPROVED** [review]
LGTM! Just a thought that a [jinja template](https://jinja.palletsprojects.com/en/2.10.x/templates/) will be very useful here and make it easier to read the actual message without all the code getting in the way (similar to the way you post all the messages in the PR description). We can revisit this if the message is getting longer and more complicated.

**Comment by janeyx99** [comment]
Comments on the actual messages to make them clearer/more useful to users:

> Force merge message:
@pytorchbot successfully started a merge job. Check the current status [here](https://github.com/pytorch/pytorch/pull/None).
The merge job was triggered with the force (-f) flag. This means your PR will be merged immediately, bypassing any CI signals.
If you have any questions or feedback, reach out to the [Pytorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)!

- Can we have the immediately be bolded? `**immediately**`
- Will there be an end message before the if you have any questions or feedback that says "if this is not your desired command, please look at the wiki for merge alternatives)? If not, we should add that message everywhere before the reach out. 
- PyTorch and not Pytorch. 

> Green Message:
@pytorchbot successfully started a merge job. Check the current status [here](https://github.com/pytorch/pytorch/pull/None).
The merge job was triggered with the green (-g) flag. This means that your PR will be merged once all signals have passed.
If you have any questions or feedback, reach out to the [Pytorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)!

- "once all signals triggered on the PR" would be clearer. 
- Could we include an eta? "once all signals triggered on the PR have passed (may take up to 4 hours depending on what is still unrun on your PR. If all signals are completely green, merge should be immediate)."
- Again, will there be an end message before the if you have any questions or feedback that says "if this is not your desired command, please look at the wiki for merge alternatives)? If not, we should add that message everywhere before the reach out. 

> Land Checks Message:
@pytorchbot successfully started a merge job. Check the current status [here](https://github.com/pytorch/pytorch/pull/None).
The merge job was triggered with the land check (-l) flag. This means that your PR will be merged once all signals on your PR have run since you have added the ciflow/trunk label to your PR.
If you have any questions or feedback, reach out to the [Pytorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)!

- there should be another land checks msg, no? One that includes the ETA? In that one, we can say "if you need to coordinate lands between different changes and cannot risk a land race, please add the `ciflow/trunk` label to your PR and wait for signal to complete, and then land your changes in proper order. Having trunk, pull, and Lint pre-run on a PR will bypass land checks and the ETA should be immediate."
- "The merge job was triggered with the land check (-l) flag. (If you did not specify the flag yourself, you are likely enrolled into land checks rollout...) This means..."

> Normal Message:
@pytorchbot successfully started a merge job. Check the current status [here](https://github.com/pytorch/pytorch/pull/None).
The merge job was triggered without a flag. Since your PR has a ciflow label, we will wait for all checks to be green.
If you have any questions or feedback, reach out to the [Pytorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)!

- With or without the ciflow label, "we will wait for all checks to be green" is just true, no? If so, let's not be confusing. 
- Also, this functionality is now the same to merge -g anyway so the descriptions should be the same. See comments above.

Generally, I realized we should also update our wikis:
- Our Bot commands one should include detailed descriptions of the accept2run and accept2ship labels if it doesn't already.
- Or DevX OH wiki should include other ways to contact us if they cannot make the OH. (e.g., tagging our team.)

**Review by janeyx99 — CHANGES_REQUESTED** [review]
I have commented things to address for each.

File: .github/scripts/trymerge_explainer.py
Comment (janeyx99) [reply .github/scripts/trymerge_explainer.py:64]: Sure, but we should just mention that they might have been here.

**Review by janeyx99 — APPROVED** [review]

**Comment by zengk95** [comment]
@pytorchbot successfully started a merge job. Check the current status [here](None).
The merge job was triggered without a flag. This means that your change will be merged once all checks on your PR have passed (ETA: 0-4 Hours). If this is not the intended behavior, feel free to use some of the other merge options in the [wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands).
Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours) with feedback or questions!

**Comment by zengk95** [comment]
@pytorchbot successfully started a merge job. Check the current status [here](None).
The merge job was triggered with the green (-g) flag. This means that your change will be merged once all checks on your PR have passed (ETA: 0-4 Hours). If this is not the intended behavior, feel free to use some of the other merge options in the [wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands).
Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours) with feedback or questions!

**Comment by zengk95** [comment]
@pytorchbot successfully started a merge job. Check the current status [here](None) and land check progress [here](https://hud.pytorch.org/pytorch/pytorch/commit/54f4f9f750a0a31712a77ee49e1fbd5a4c15f43c)..
The merge job was triggered with the land checks (-l) flag. If you did not specify this flag yourself,  you are likely enrolled in the [land checks rollout](https://github.com/pytorch/test-infra/blob/main/torchci/lib/bot/rolloutUtils.ts#L1-L34). This means that your change will be merged once all checks on your PR and the land checks have passed (**ETA 4 Hours**). If you need to coordinate lands between different changes and cannot risk a land race, please add the `ciflow/trunk` label to your PR and wait for signal to complete, and then land your changes in proper order. Having `trunk`, `pull`, and `Lint` pre-run on a PR will bypass land checks and the ETA should be immediate. If this is not the intended behavior, feel free to use some of the other merge options in the [wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands).
Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours) with feedback or questions!

**Pull request closed without merging**
